proxmoxtf/resource/vm.go

```diff
@@ -1502,6 +1514,78 @@ func vmCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.D
 	return vmCreateCustom(ctx, d, m)
 }
 
+// Check for an existing CloudInit IDE drive. If no such drive is found, return -1.
+func findExistingCloudInitDrive(vmConfig *vms.GetResponseData, vmID int, defaultValue string) string {
+	devices := []*vms.CustomStorageDevice{
+		vmConfig.IDEDevice0, vmConfig.IDEDevice1, vmConfig.IDEDevice2, vmConfig.IDEDevice3,
+	}
+	for i, device := range devices {
+		if device != nil && device.Enabled && device.Media != nil && *device.Media == "cdrom" && strings.Contains(
+			device.FileVolume,
+			fmt.Sprintf("vm-%d-cloudinit", vmID),
+		) {
+			return fmt.Sprintf("ide%d", i)
```

Comment:
Technically, `device.Interface` _should_ return the proper value, but I also see the point in hardcoding "ideN" in here and not to deal with potential `nil`s.

Reply:
I didn't know about `device.Interface` ! I would have used that if I had. However your point about potential `nil`s in data which comes from Proxmox and on which the provider has no control is very valid.

Your pointing me at this chunk of code let me see a mistake in the comment for this function though. I will update this shortly.